### DISCUSSION TITLE
Behavior for downcasting to numpy with reducable units (like mm/m)

### DISCUSSION THREAD

**Discussion by knightofni-i** [hgrecco/pint#1756] [Q&A] [answered]
How hard would it be to attempt to cast reducable units to dimensionless before stripping units when downcasting to numpy? Has this been considered and omitted or just not yet implemented? If the latter, I can do some digging to see how hard it would be to implement.

`>>> aovert = UREG.Quantity([100, 200, 400], "mm") / UREG.Quantity(1, "m")`
`>>> np.array((.1, 2)).searchsorted(aovert)`
`array([2, 2, 2], dtype=int64)`
A UnitStrippedWarning is also issued.

Whereas we would expect the result yielded by:
`>>> np.array((.1, 2)).searchsorted(aovert.m_as(UREG.dimensionless))`
`array([0, 1, 1], dtype=int64)`

**Comment by andrewgsavage** [comment, marked as answer]
You probably want to turn auto_reduce_dimensions on

https://pint.readthedocs.io/en/stable/tutorial.html
> If you want pint to automatically perform dimensional reduction when producing new quantities, the UnitRegistry class accepts a parameter auto_reduce_dimensions. Dimensional reduction can be slow, so auto-reducing is disabled by default.
